x-pack/platform/plugins/shared/streams/server/lib/telemetry/stats/utils.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IngestStreamLifecycle } from '@kbn/streams-schema';
+
+/**
+ * Determines if a stream has changed retention from the default.
+ *
+ * @param lifecycle - The stream's lifecycle configuration
+ * @returns true if retention is changed from default, false otherwise
+ *
+ * Simple logic:
+ * - inherit: {} = Default retention (false)
+ * - dsl: {} = Forever retention (true)
+ * - dsl: { data_retention: "30d" } = Custom retention (true)
+ * - ilm: { policy: "policy" } = ILM retention (true)
```

Comment:
This isn't what the function is actually doing, but it's what it _should_ be doing. Changing lifecycle to another ILM policy should totally count here.

The PR description is not right:

> Number of classic streams with custom retention settings (DSL or ILM). Counted by lifecycle containing "dsl" property (includes custom periods and "Forever" retention).

Reply:
Yes, it needed fixing, I've improved it.